packages/patrol/android/src/main/kotlin/pl/leancode/patrol/Automator.kt

```diff
@@ -204,24 +204,36 @@ class Automator private constructor() {
 
         val uiSelector = UiSelector().className(EditText::class.java).instance(index)
         val uiObject = uiDevice.findObject(uiSelector)
-        uiObject.click()
+
+        if (showKeyboard) {
+            uiObject.click()
+        }
+
         uiObject.text = text
 
-        pressBack() // Hide keyboard.
+        if (showKeyboard) {
+            pressBack() // Hide keyboard.
+        }
     }
 
-    fun enterText(text: String, uiSelector: UiSelector, bySelector: BySelector, index: Int) {
+    fun enterText(text: String, uiSelector: UiSelector, bySelector: BySelector, index: Int, showKeyboard: Boolean) {
         Logger.d("enterText($text): $uiSelector, $bySelector")
 
         if (waitForView(bySelector, index) == null) {
             throw UiObjectNotFoundException("$uiSelector")
         }
 
         val uiObject = uiDevice.findObject(uiSelector).getFromParent(UiSelector().className(EditText::class.java))
```

Comment:
Body of this method is the same for both cases. The only difference is finding this `uiObject`. Could you move common entering text & keyboard logic into a private method? common for those two?

Reply:
I'm a follower of https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

Reply:
This code isn't the same by accident. It's the same on purpose, because those methods differ only in getting the element. Rest must be the same. Why not express this in code?

Reply:
I understand, but I just don't agree. 

My other argument is to make Patrol's iOS and Android code look similar to each other (to make changes easier), and on iOS these methods do have a much different implementation – so it makes sense to separate them there.

Reply:
But I appreciate your comment - thank you